### ISSUE TITLE
doxmlparser can't properly parse enumerated types

### ISSUE THREAD

**Issue by dhowland** [doxygen/doxygen#9802]
**Describe the bug**
The generated code that implements doxmlparser apparently has a bug.

When parsing enumerated value types, the `name` property is always `None`.  In other words, you can't read the enumerated values.

I have fixed the bug in the following commit to my fork, here: 19d77e6ce50ccbf3877af9fc892b0f33c373569c

That commit log contains further context.  Given that this is generated code, I can't send a pull request.  Honestly I'm not sure if this is something that is fixable, but I discovered the issue and fixed it, so I'm making a public record.

**Expected behavior**
The `name` property should be populated.

**To Reproduce**
See attached demonstration.
[bugdemo.zip](https://github.com/doxygen/doxygen/files/10469658/bugdemo.zip)

**Version**
Doxygen 1.9.5
Python 3.10

**Comment by albert-github** [comment]
I'm not sure about this, looking at the generated code we see:
```
    def buildChildren(self, child_, node, nodeName_, fromsubclass_=False, gds_collector_=None):
        if nodeName_ == 'name' and child_.text is not None:
            valuestr_ = child_.text
            valuestr_ = self.gds_parse_string(valuestr_, node, 'name')
            valuestr_ = self.gds_validate_string(valuestr_, node, 'name')
            obj_ = self.mixedclass_(MixedContainer.CategorySimple,
                MixedContainer.TypeString, 'name', valuestr_)
            self.content_.append(obj_)
            self.name_nsprefix_ = child_.prefix
```

and in the `class MixedContainer`:
```
    def __init__(self, category, content_type, name, value):
        self.category = category
        self.content_type = content_type
        self.name = name
        self.value = value
    def getCategory(self):
        return self.category
    def getContenttype(self, content_type):
        return self.content_type
    def getValue(self):
        return self.value
    def getName(self):
        return self.name
```
so wouldn't it be possible  to obtain the "value" of the "name" in this way?

**Comment by dhowland** [comment]
Well there is no public getter method that produces `content_`.  However the implementation of `bugdemo.py` would look like this:

'''
for compounddef in root.get_compounddef():
    for sectiondef in compounddef.get_sectiondef():
        if sectiondef.get_kind() == DoxSectionKind.ENUM:
            for memberdef in sectiondef.get_memberdef():
                for enumvaluedef in memberdef.get_enumvalue():
                    for item in enumvaluedef.content_:
                        if item.getName() == 'name':
                            print(memberdef.get_name(), item.getValue())
'''

It does produce the expected results.  I don't really know how generateDS works, so I admit freely that I might be totally wrong.  However by the what I see in `compound.xsd`, I'm fairly sure that the original implementation is "correct".  Compare to how `memberdefType` works.

Like I said, this is probably only be fixable in the generateDS project, thus not your problem.  It just seems like an issue to me.

**Comment by dhowland** [comment]
Okay I fiddled with generateDS a little bit.

I "fixed" `compound.xsd` here: be1453909df1886aec93193c931874e0d71076cb

**Comment by albert-github** [comment]
Looks indeed  a valid fix. When looking at the documentation of `mixed` at https://www.davekuhlman.org/generateDS.html#mixed-content we see:

> 9.2   Mixed content
>
> Elements that are defined to contain both text and nested child elements have "mixed content". `generateDS.py` provides access to mixed content, but the generated data structures (classes) are fundamentally different from that generated for other elements. See section [Mixed content](https://www.davekuhlman.org/generateDS.html#mixed-content-1) for more details.
> 
> Note that elements defined with attributes but with no nested sub-elements do not need to be declared as "mixed". For these elements, character data is captured in a member variable `valueOf_`, and can be accessed with member methods `getValueOf_` and `setValueOf_`.

**Comment by doxygen** [comment]
Indeed, this should not be a mixed type.

**Comment by dhowland** [comment]
Created #9809

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
